### PR TITLE
fix(TPG >= 4.53)!: update regional proxy lb module

### PR DESCRIPTION
**Pull request by MahekKatariya**
### Pull Request Updates

This PR introduces several improvements to enhance flexibility and user experience when creating resources in Google Cloud using Terraform. The following changes have been made:

1. **Default Value for `target_tags`:**
   - A default value of `[]` has been added to the `target_tags` variable. This allows users to omit this value  if they do not wish to create firewall rules, simplifying the configuration.

2. **Default Value for `proxy_only_subnet_cidr`:**
   - A default value of `""` has been added to the `proxy_only_subnet_cidr` variable. Users can now omit this value if they do not wish to create a proxy-only subnet, providing more flexibility in subnet configuration.

3. **New `proxy_header` Parameter for `google_compute_region_target_tcp_proxy`:**
   - A new optional parameter, `proxy_header`, has been introduced in the `google_compute_region_target_tcp_proxy` resource. Users can now specify a value for `proxy_header` if desired. The default value is set to `NONE`.

These changes make it easier for users to create custom configurations without unnecessary overhead, while also providing sensible defaults.

### CODE REVIEW

File: modules/regional_proxy_lb/variables.tf [hunk 1]
```diff
@@ -66,6 +67,7 @@ variable "create_proxy_only_subnet" {
 
 variable "proxy_only_subnet_cidr" {
   type        = string
+  default     = ""
```
Comment (imrannayer) [inline modules/regional_proxy_lb/variables.tf:70]: @MahekKatariya can u plz add description that this is required if `create_proxy_only_subnet` is set to true.

Comment (MahekKatariya) [reply modules/regional_proxy_lb/variables.tf:70]: @imrannayer updated the description

**Comment by MahekKatariya** [comment]
> @MahekKatariya thanks for the PR. Can you plz update version.tf of regional proxy load balancer. Change minimum provider required to 4.53 as this feature was part of [4.53 release](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4530-february-13-2023)

@imrannayer Changed minimum version to 4.53

**Review by imrannayer — APPROVED** [review]

**Pull request merged**